rectools/models/pure_svd.py

```diff
@@ -78,6 +93,15 @@ def __init__(
         self.tol = tol
         self.maxiter = maxiter
         self.random_state = random_state
+        if use_gpu:  # pragma: no cover
+            if not HAS_CUDA:
```

Comment:
I think it's better to use `cp.cuda.runtime.getDeviceCount() == 0` instead:
1. To achieve full compatibility and do not depend on implicit
2. There is an issue that on some machines with GPU and Cuda available, and e.g. torch is working, `HAS_CUDA` is still `False`

Reply:
Okay, I'll use `cp.cuda.runtime.getDeviceCount() == 0` instead.

I aware that `recommend_use_gpu_ranking` depends on `HAS_CUDA`, so we may have room for consistency.
https://github.com/MobileTeleSystems/RecTools/pull/229/files#diff-95c3d16fac303204b014a3e5359eb37c595633c1950b7e5f31738a819717dbe0R81

Reply:
Fixed 6c060fbf6f266e9c7347c28431d7c3ae144e6ffa

Reply:
`recommend_use_gpu_ranking` uses the `recommend` method from `implicit`, if `implicit` doesn't see Cuda, it won't work on GPU, and here it may still work if `cupy` sees Cuda (also vice versa - if implicit sees cuda and cupy doesn't)